### ISSUE TITLE
[mu4e bug] (re)indexing never stops when store is empty

### ISSUE THREAD

**Issue by vchouraki** [djcb/mu#2240]
**Describe the bug**
On a completely empty store that's not supposed to have new messages after updating, starting mu4e (or updating with U) causes the echo area to display the following message forever:

> [mu4e] Store is empty; (re)indexing. This may take a while.

The message buffer displays the following messages in a loop:

> [mu4e] Store is empty; (re)indexing. This may take a while.
> [mu4e] Indexing... processed 0, updated 0
> [mu4e] Indexing completed; processed 0, updated 0, cleaned-up 0

Mu4e is then mostly unusable as some shortcuts and point seem to be resetted with each "loop". Quitting makes it impossible to restart mu4e unless mu server process is killed in the terminal (similar to #2198). 

I can get out of this loop by hitting "H" on the home screen. Then, I can quit mu4e and restart it without having to kill the mu server process.

Adding a single mail anywhere in the store prevents this.

**To Reproduce**
Start mu4e on an empty store

**Environment**
Ubuntu 18.04
isync 1.4.4

mu version                           : 1.6.10

Xapian version                       : 1.4.5
GLib version                         : 2.56.4
GMime version                        : 3.2.0
Guile version                        : 2.2.3
Build mu4e                           : yes
Emacs version                        : GNU Emacs 29.0.50
Build 'mug' toy-ui (gtk+/webkit)     : no

Have direntry->d_ino                 : yes
Have direntry->d_type                : yes

**Checklist**
- [X] you are running vanilla emacs (i.e. without Doom, Evil, Spacemacs etc.) (otherwise, please try to reproduce without those
- [X] you are running mu4e without any third-party extensions (otherwise, please try to reproduce without those)
- [X] you are running either the latest 1.6.x release, or a 1.7.x development release (otherwise, please upgrade).

**Comment by djcb** [comment]
This works for me:
```
Sources/mu % mkdir /tmp/test
Sources/mu % mu/mu mkdir /tmp/test/maildir
Sources/mu % mu/mu init --muhome=/tmp/test --maildir=/tmp/test/maildir
maildir           : /tmp/test/maildir
database-path     : /tmp/test/xapian
schema-version    : 452
max-message-size  : 100000000
batch-size        : 250000
messages in store : 0
created           : 2022-04-30T01:51:14 EEST
personal-address  : <none>

store created; use the 'index' command to fill/update it.
Sources/mu % mu/mu index --muhome=/tmp/test
indexing maildir /tmp/test/maildir -> store /tmp/test/xapian
| indexing messages; processed: 0; updated/new: 0; cleaned-up: 0
Sources/mu %                                                      
```
this is with the latest in the release-branch, so a  little bit beyond 1.6.10 (and indeed there has been an indexing update recently). Can you test? If it works, I can roll a 1.6.11.

Thanks.

**Comment by vchouraki** [comment]
Thanks for your reply.

I think mu is working fine when called from the CLI : I ran the commands you provided and got the same output as you.

Unfortunately, the problem remains when calling mu4e in emacs. I also tried with a minimal mu4e config starting from "emacs -Q" and got the same results. I also tried to build mu from the latest commit using an Ubuntu 22.04 VM, also with the same results (could not build it in ubuntu 18.04 as glib version is too old).

HTH.
